### PR TITLE
Add package size limit to FAQ

### PR DESCRIPTION
**Pull request by maartenba**
@yishaigalatzer @harikm86

### CODE REVIEW

File: NuGet.Docs/Consume/NuGet-FAQ.md [hunk 1]
```diff
@@ -212,6 +212,15 @@ We encourage the NuGet community to work together to resolve any disputes that m
 
 For test purposes, you can use [staging.nuget.org](http://staging.nuget.org), or alternative public nuget servers like https://myget.org (public and private feeds) or [Visual Studio Online](https://blogs.msdn.microsoft.com/visualstudioalm/2015/08/27/announcing-package-management-support-for-vsotfs/) . Please note that the packages being uploaded to staging.nuget.org may not be preserved. More details can be found [here.](http://blog.nuget.org/20130419/goodbye-preview.html)
 
+**What is the maximum size of packages I can upload to NuGet.org?**
+
+NuGet.org allows uploading packages up to 250MB. However, we recommend uploading smaller packages that are < 1 MB and using dependencies to link together packages.
```
Comment (harikm86) [inline NuGet.Docs/Consume/NuGet-FAQ.md]: < 1MB. is this a number we have come to based on some analysis?

File: NuGet.Docs/Consume/NuGet-FAQ.md [hunk 2]
```diff
@@ -212,6 +212,15 @@ We encourage the NuGet community to work together to resolve any disputes that m
...
+
+NuGet's protocol for downloading packages is HTTP. The bigger the package, the higher the chance for failed installs. It is much easier to retry and download small components than large ones.
```
Comment (harikm86) [inline NuGet.Docs/Consume/NuGet-FAQ.md]: Lets use packages vs components consistently

File: NuGet.Docs/Consume/NuGet-FAQ.md [hunk 3]
```diff
@@ -212,6 +212,15 @@ We encourage the NuGet community to work together to resolve any disputes that m
...
+
+It is possible to share dependencies between multiple packages, making the total download size for consumers of your NuGet packages smaller.
+
+Many times the dependency is mostly static and never change. When fixing a bug in code, the dependencies may not have to be updated. If your dependencies ship in the same package you end up reshipping the heavy components again and again. By splitting NuGet packages in dependencies, upgrades are much more fine-grained for the end user of the NuGet package.
```
Comment (harikm86) [inline NuGet.Docs/Consume/NuGet-FAQ.md]: change Many times to In most cases,

Comment (harikm86) [inline NuGet.Docs/Consume/NuGet-FAQ.md]: instead of heavy lets use large

Comment (harikm86) [inline NuGet.Docs/Consume/NuGet-FAQ.md]: Instead of again and again lets use every time.

Comment (harikm86) [inline NuGet.Docs/Consume/NuGet-FAQ.md]: By splitting NuGet packages into smaller chunks,

**Comment by harikm86** [comment]
Minor comments above

File: NuGet.Docs/Consume/NuGet-FAQ.md (hunk 1 above)
Comment (maartenba) [inline NuGet.Docs/Consume/NuGet-FAQ.md]: It's more or less a common best practice. @yishaigalatzer

**Comment by maartenba** [comment]
Updated

**Comment by yishaigalatzer** [comment]
Lgtm

**Comment by yishaigalatzer** [comment]
I would only say that the 1mb is a best practice, and that many times dependencies are static. 

I would also add that you don't to include other dlls in a package to avoid collisions so for dotnet the rule of thumb is 1  dll per package

File: NuGet.Docs/Consume/NuGet-FAQ.md [hunk 4]
```diff
@@ -212,6 +212,15 @@ We encourage the NuGet community to work together to resolve any disputes that m
...
 
+**What is the maximum size of packages I can upload to NuGet.org?**
+
+NuGet.org allows uploading packages up to 250MB. However, we recommend uploading smaller packages that are < 1 MB and using dependencies to link together packages.
+
+NuGet's protocol for downloading packages is HTTP. The bigger the package, the higher the chance for failed installs. It is much easier to retry and download small packages than large ones.
+
+It is possible to share dependencies between multiple packages, making the total download size for consumers of your NuGet packages smaller.
+
+In most cases the dependency is mostly static and never change. When fixing a bug in code, the dependencies may not have to be updated. If your dependencies ship in the same package you end up reshipping the large packages every time. By splitting NuGet packages into smaller chunks (dependencies), upgrades are much more fine-grained for the end user of the NuGet package.
```
Comment (harikm86) [inline NuGet.Docs/Consume/NuGet-FAQ.md]: dependencies are mostly static and never changes

**Comment by harikm86** [comment]
One last comment and signed off :)

**Comment by maartenba** [comment]
Updated

**Comment by harikm86** [comment]
Looks good to me. Lets get it in!

**Pull request merged**